enterprise/coordinator/history/configmapstore.go

```diff
@@ -0,0 +1,225 @@
+// Copyright 2025 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+//go:build enterprise
+
+package history
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+)
+
+const timeout = 10 * time.Second
+
+var keyRe = regexp.MustCompile(`^[a-zA-Z0-9-]+/[a-zA-Z0-9-]+$`)
+
+// ConfigMapStore is a Store implementation backed by Kubernetes Config Maps.
+type ConfigMapStore struct {
+	client    kubernetes.Interface
+	namespace string
+	uid       types.UID
+	logger    *slog.Logger
+}
+
+// NewConfigMapStore creates a new instance backed by Kubernetes Config Maps.
+func NewConfigMapStore(client kubernetes.Interface, log *slog.Logger) (*ConfigMapStore, error) {
+	namespace, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	if err != nil {
+		return nil, err
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	self, err := client.CoreV1().Pods(string(namespace)).Get(ctx, os.Getenv("HOSTNAME"), metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	ownerRefs := self.GetOwnerReferences()
+	var statefulset metav1.OwnerReference
+	for _, ownerRef := range ownerRefs {
+		if ownerRef.Kind == "StatefulSet" && ownerRef.Name == "coordinator" {
+			statefulset = ownerRef
+			break
+		}
+	}
+	if statefulset.UID == "" {
+		return nil, fmt.Errorf("coordinator statefulset not found")
+	}
+	return &ConfigMapStore{
+		client:    client,
+		namespace: self.Namespace,
+		uid:       statefulset.UID,
+		logger:    log,
+	}, nil
+}
+
+// Get the value for key.
+func (s *ConfigMapStore) Get(key string) ([]byte, error) {
+	if !keyRe.MatchString(key) {
+		return nil, fmt.Errorf("invalid key %q", key)
+	}
```

Comment:
It could make sense to merge `objectName` and the check, since you're destructuring the string the same way in both functions. And the resulting function could have a nice little unit test.